### ISSUE TITLE
On Hold Status & Custom Status

### ISSUE THREAD

**Issue by supportreq** [ladybirdweb/faveo-helpdesk#91]
On Hold Status which will pause the SLA timer in case the item is dependent on a third party vendor...

Also custom Statuses to allow users to add more customized statuses for better reporting and workflow management..

**Comment by mbiolsi** [comment]
Custom status would be GREAT! We use status like "Pending Client" and "Pending 3rd Party" all of the time in our other software because it helps us quickly see who needs to be working on it.

**Comment by mbiolsi** [comment]
NOTE: after looking at the roadmap: if custom tags are implemented it could possibly eliminate the need for custom status.

**Comment by bhanu2217** [comment]
We are already working on custom status, it should be released by end of this month.
We will  be adding tags also in the future.

**Comment by bhanu2217** [comment]
**Requirement**

**Admin Panel:**
In Custom status CRUD add checkbox, Pause SLA. So we know for which status SLA is enabled and for which SLA is disabled

Whenever status of particular ticket is changed, Status SLA Module(Class) to check if for this particular Status we need to enable or disable SLA timer is called.

**DB:**
- Put a boolean field SLA enable/disable in ticket table
- Put a date time field to enter the date time when SLA was stopped and enabled in ticket table

**Scenario 1: Status choose in which SLA is stopped**
- For this particular ticket enable SLA stop timer and put date time in the DB

**Scenario 2: Status choose in which SLA is enabled and last status SLA was stoped**
- Calculate how much SLA time was remaining since SLA was stopped.
- Add this particular remaining time to SLA due date from now based on business hours and Holiday lust
- For this particular ticket enable SLA start timer and put date time in the DB

**Ticket Activity:**
Update ticket activity whenever SLA is enabled or disabled

**Ticket reply box:**
Below the ticket reply box put a dropdown where along with the reply ticket status can be changed.

**Comment by webaxys** [comment]
Hello, 

I was told on the site live chat, this was operational... but from my tests it does not occur correctly (on the live demo)

I created a SLA PLAN with 4 hours resolve time. 
I've used "stop sla" , then 2 cases:

* if I replied, then the due date is back to 4hrs from the ticket creation... I expected : 4hrs from the ticket creation + the time I stopped it...

* if I edit to come back to "open" status again: then it uses 10hrs SLA from the ticket creation which comes from I dont know where....

I'm seeking a system able to pause/resume sla and which counts the "paused" time in due time... (which looks like scenario2)

so I'm asking if this is implemented or yet to be implemented... This is crucial to me :)
thanks a lot

**Comment by vijaysebastian** [comment]
@webaxys  halting SLA stops the 'due date' calculation; That means if you stop the sla, due date become null, then if you change the status to any 'open' (purpose of the status) status, it re-calculate the due date from created ticket date. This is how it is working in current system.

**Comment by webaxys** [comment]
thanks a lot Sebastian, that's also what I stated... but I lost  2 hrs being told it was working like said in bhanu2217 's last reply. So tried again and again, for nothing :p
that's a function I'm looking for desperately, has someone any ETA for when it would be released?

**Comment by vijaysebastian** [comment]
@webaxys we are not calculating the paused time. I have a doubt about adding paused time, then what is the use of stopping sla ! Can you please elaborate more ? If it is generic, I can discuss with in my team and we include in any upcoming releases.

**Comment by webaxys** [comment]
well, in some activities, we have SLA contracts with clients but in some emergency/force majeure cases , the SLA is going to pause because responsability is on 3rd parties (like government or public instances...) that are not submitted to our own SLA terms.
that's why it would "pause" or "freeze" the SLA until it is resumed.

I saw many threads on this kind of request amongst many helpdesk products, though rarely implemented. Some other cases mentioned by people speak about freezing SLA when depending on 3rd parties like furnishers , due to the time they receive an hardware piece for example.
could be a plus for faveo ;)

**Comment by ssmittal** [comment]
In Pro edition we have stop SLA linked to status, you can create as many status with different name/words for halting the SLA.

Currently the system doesn't have explicit button to start or stop SLA, starting and stopping of SLA is done via ticket status.

Community edition doesn't have advance SLA's linked to business house, holiday list or ability to start/stop SLA.

To solve your problem in PRO edition, you can create halt SLA status something like **Government Dependent** or **Waiting for vendor response** and this should take care of what you are looking for

**Comment by webaxys** [comment]
No, indeed, I wasn't specific enough, we need to resume SLA Due Time from where it stopped.
I sent an email to sales department for getting a quotation on how much it would cost for such a specific development.

**Comment by vijaysebastian** [comment]
@webaxys We have updated the Halt SLA feature, according to feedback, and it will release on this week.

We have different level of reporting; For your understanding, i am including some screen shots

![screen shot 2017-03-20 at 4 35 37 pm](https://cloud.githubusercontent.com/assets/10737619/24097325/4e4ce864-0d8b-11e7-8a0a-27c857783e42.png)

This gives you overall ticket details grouped by status, priority, source and type

![screen shot 2017-03-20 at 4 37 59 pm](https://cloud.githubusercontent.com/assets/10737619/24097412/b2075178-0d8b-11e7-8444-1b3a40eec33a.png)

This report shows the trend of your issue traffic and you can track by weekly, monthly and yearly basis 

![screen shot 2017-03-20 at 4 38 31 pm](https://cloud.githubusercontent.com/assets/10737619/24097460/e701e6b8-0d8b-11e7-8257-dcbb5d892061.png)

From this report you can find out the traffic of days(mon to sun) and hours


![screen shot 2017-03-20 at 4 43 08 pm](https://cloud.githubusercontent.com/assets/10737619/24097548/53bab7ee-0d8c-11e7-8de0-06d7f457e708.png)


From here you can see the department associated with issues. And you can export to csv file. Same for agent performance and team performance

![screen shot 2017-03-20 at 4 45 28 pm](https://cloud.githubusercontent.com/assets/10737619/24097610/a7a2c3ba-0d8c-11e7-8aab-fa1685ceaf31.png)

Here you can see the response time and resolve time, categorised with time consumption 

![screen shot 2017-02-23 at 11 50 12 am](https://cloud.githubusercontent.com/assets/10737619/24097683/fbe04de4-0d8c-11e7-843e-b36b99f0d28c.png)
 
For billable tickets, you can export and mail to anyone in the system 

We have added two more reports for organisation based ticket distribution and Report generated from customer's rating   


These all features we have included in our upcoming release. Probably, this week. We are doing final round of testing.

**Comment by webaxys** [comment]
hello,

thanks for your feedback...
I tried the online demo and havent been able to see what's different on an operational point of view.
I created a new status which has "sla stopped" set to "yes".
when i apply it, the due date disappeared, ok.
when i put it back to open, the SLA date is still set to where it was before stopped (ex: sla set at 15h00, defining resolution to 19h00. stopped during 30mins, still showing 19h00 after reactivation...

i feel like I'm misunderstanding something probably...

**Comment by PoonamPrusty** [comment]
Hi,

Thanks for evaluating Face help desk.

I guess you are confused with the SLA halt option.

On choosing to halt, the SLA calculation stops and resumes when you unhalt
the SLA. It is helpful when you know the time taken to solve the issue
would be more than the SLA time set.

That way, the ticket won't overdue and will stay active until the issue is
resolved.

Let me know if you are still not able to understand.

**Comment by webaxys** [comment]
hello, 
thanks for your message,
so it's not "halting the sla" , it's just disabling it, and disabling the overdue time...
just correct me if I'm wrong: it's not stopping and resuming the sla, like you would stop and resume a chronometer... which was what was originally discussed on this thread...  ;)

brest regards :)
